src/main/fox_demo.c

```diff
@@ -0,0 +1,3005 @@
+        0.0f,
+        -250.0f,
+    },
+};
+
+f32 D_800CA170[] = {
+    0.150000006f, 0.4499999881f, 0.75f, 1.299999952f, 0.75f, 0.4499999881f, 0.150000006f, 0.0f,
+};
+
+f32 D_800CA190[] = {
+    5.0f,
+    5.199999809f,
+};
+
+f32 D_800CA198[] = {
+    0.5f, 1.5f, 2.5f, 1.5f, 0.5f, 0.200000003f, 0.200000003f,
+};
+
+f32 D_800CA1B4[] = {
+    -30.0f, -60.0f, -90.0f, -120.0f, -150.0f, -180.0f, -200.0, -210.0,
+};
+
+f32 D_800CA1D4[] = {
+    1.5f, 0.8700000048f, 0.8299999833f, 0.75f, 0.6999999881f, 0.6000000238f, 0.400000006f, 0.1000000015f,
+};
+
+extern AnimationHeader* D_800CA1F4[] = {
+    D_60265B4, D_602B8DC, D_60034C4, D_602A2CC, D_602CEB4, D_602B778, (AnimationHeader*) 0x0601F3B8,
+};
+
+extern void func_80187520(s32, void*);
+extern void func_800AB334(void);
+extern void func_800ADF58(Player*);
+extern void func_8004D440(Player*);
+extern void func_80046358(Player*);
+extern void func_8004B368(Player*);
+extern void func_80095604(Player*);
+extern void func_800B2130(Player*);
+extern void func_801882CC(Player*, OSContPad**, OSContPad**);
+extern void func_8018E084(Player*);
+extern void func_8018F94C(Player*);
+extern void func_80191160(Player*);
+extern void func_80193C4C(Player*);
+extern void func_80194728(Player*);
+extern void func_80196D88(Player*);
+extern void func_80197290(Player*);
+extern void func_8019C85C(Player*);
+extern void func_8019D76C(Player*);
+extern void func_8019EE60(Player*);
+extern void func_801A7930(Player*);
+extern void func_801AF8F4(Player*, OSContPad**, OSContPad**);
+extern void func_801B3D04(Player*, OSContPad**, OSContPad**);
+extern void func_8002ED60(Player*);
+extern void func_800935E8(Player*);
+extern void func_8018769C(Player*);
+extern void func_8018CD8C(Player*);
+extern void func_8018DF74(Player*);
+extern void func_8018EF6C(Player*);
+extern void func_8018F880(Player*);
+extern void func_80192E20(Player*);
+extern void func_80193A30(Player*);
+extern void func_80196BF8(Player*);
+extern void func_801988B8(Player*);
+extern void func_8019BA64(Player*);
+extern void func_8019D428(Player*);
+extern void func_801A0AC0(Player*);
+extern void func_801A10F4(Player*);
+extern void func_801AB9B0(Player*);
+extern void func_801AC754(Player*);
+extern void func_800A46A0(Player*);
+extern void func_800AD7F0(Player*);
+extern void func_800B2574(Player*);
+extern void func_800AE278(Player*);
+extern void func_80093164(Object_2F4*);
+extern void func_800A6028(Vec3f*, u32);
+extern void func_8018DA58(Object_2F4*);
+extern void func_8018ED9C(Object_2F4*);
+extern void func_80195E44(Object_2F4*);
+extern void func_80197F10(Object_2F4*);
+extern void func_8019DD20(Object_2F4*);
+extern void func_8019E5F0(Object_2F4*);
+extern void func_8019FF00(Object_2F4*);
+extern void func_801A8BE8(Object_2F4*);
+extern void func_801B28BC(Object_2F4*);
+
+void func_80048AC0(s32 arg0) {
+    s32 teamShield;
+
+    if (arg0 == 1) {
+        if (gTeamShields[1] < 0x40) {
+            func_800BA808(gMsg_ID_20303, 0xA);
```

Comment:
The second argument of `func_800BA808` uses the enum `RadioCharacterId`. In fact, its argument should probably use that as its type. Could you change it in fox_play and functions.h?